source/common/stats/symbol_table_impl.h

```diff
@@ -128,12 +134,33 @@ class StatNameImpl : public StatName {
   ~StatNameImpl() override { symbol_table_.free(symbol_vec_); }
   std::string toString() const override { return symbol_table_.decode(symbol_vec_); }
 
+  // Returns a hash of the underlying symbol vector, since StatNames are uniquely defined by their
+  // symbol vectors.
+  uint64_t hash() const override { return HashUtil::hashVector(symbol_vec_); }
+  // Compares on the underlying symbol vectors.
+  // NB: operator==(std::vector) checks size first, then compares equality for each element.
+  bool operator==(const StatName& rhs) const override {
+    const StatNameImpl& r = dynamic_cast<const StatNameImpl&>(rhs);
```

Comment:
Are there multiple impls of StatName?

Maybe you could just define `operator==(const StatNameImpl&)` and make the map use StatNameImpl as well?

Reply:
Good call -- there are no other implementations of either StatName or SymbolTable. To avoid this problem, I'll just define `operator==(const StatNameImpl&)` and we can skip this whole `dynamic_cast` business.

Reply:
OK, got rid of the pure interfaces in cbf710b.